src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer_experimental.php

```diff
@@ -0,0 +1,220 @@
+use Symfony\Component\Serializer\Serialize\Template\NormalizerEncoderTemplateGenerator;
+use Symfony\Component\Serializer\Serialize\Template\Template as SerializeTemplate;
+use Symfony\Component\Serializer\Template\TemplateVariationExtractor;
+use Symfony\Component\Serializer\Template\TemplateVariationExtractorInterface;
+use Symfony\Component\Serializer\Type\ReflectionTypeExtractor;
+use Symfony\Component\Serializer\Type\TypeExtractorInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->parameters()
+        ->set('.serializer.cache_dir.template', '%kernel.cache_dir%/serializer/template')
+        ->set('.serializer.cache_dir.lazy_ghost', '%kernel.cache_dir%/serializer/lazy_ghost')
+    ;
+
+    $container->services()
+        // Serializer/Deserializer
+        ->set('serializer.serializer', Serializer::class)
+            ->args([
+                service('serializer.serialize.template'),
+                abstract_arg('serialize/deserialize runtime services'),
+                param('.serializer.cache_dir.template'),
+            ])
+        ->alias(SerializerInterface::class, 'serializer.serializer')
+
+        ->set('serializer.deserializer', Deserializer::class)
+            ->args([
+                service('serializer.deserialize.template'),
+                abstract_arg('serialize/deserialize runtime services'),
+                service('serializer.instantiator'),
+                param('.serializer.cache_dir.template'),
+            ])
+        ->alias(DeserializerInterface::class, 'serializer.deserializer')
+
+        // Template
+        ->set('serializer.serialize.template', SerializeTemplate::class)
+            ->args([
+                service('serializer.template_variation_extractor'),
+                service('serializer.serialize.data_model_builder'),
+                abstract_arg('serialize template generators'),
+                param('.serializer.cache_dir.template'),
+            ])
+
+        ->set('serializer.deserialize.template', DeserializeTemplate::class)
+            ->args([
+                service('serializer.template_variation_extractor'),
+                service('serializer.deserialize.data_model_builder'),
+                abstract_arg('deserialize template generators'),
+                param('.serializer.cache_dir.template'),
+                param('serializer.lazy_deserialization'),
+            ])
+
+        // Template variations
+        ->set('serializer.template_variation_extractor', TemplateVariationExtractor::class)
+        ->alias(TemplateVariationExtractorInterface::class, 'serializer.template_variation_extractor')
+
+        // Template generators
+        ->set('serializer.serialize.template_generator.json', SerializeJsonTemplateGenerator::class)
+            ->tag('serializer.serialize.template_generator', ['format' => 'json'])
+
+        ->set('serializer.serialize.template_generator.csv', NormalizerEncoderTemplateGenerator::class)
```

Comment:
are you planning to add XML at some point?

Reply:
Yes, it is possible to add it. But XML is a weird format (with a tree shape, attributes, etc.). Therefore, I wanted to be sure that it'll be needed before implementing it. WDYT?

Reply:
As a person who works with projects dealing with huge xmls, I can't see how a new serializer system would not support xmls 🤷‍♂️

But I can understand the complexity involved with xmls!

Reply:
Please don’t. XML is better handled by DOM or SAX parsers and is natively supported by PHP.

XML support in the current version of the Serializer has always been broken (or at least very very limited), hard to maintain, and almost useless except for very basic use cases. XML is way more complex than JSON, YAML and CSV (DTD etc) and requires dedicated tooling. It’s almost always an error to have a one to one mapping between an XML document and a PHP object.

Reply:
ok I kinda agree with that, but shouldn't we get feature parity at some point between "new" and "old" serializer systems?

I know there is a trade-off to make, and maybe all features should not be ported, but the need to parse xml and transform it to a POPO is still there, and IMO deserves an easy way to do it, even if it leverages some other components under the hood

Reply:
I replied here: https://github.com/symfony/symfony/pull/51718#issuecomment-1734545477